### PR TITLE
chore(import-export): Update import reducer to ts

### PR DESCRIPTION
**Pull request by Anemy**
Part of COMPASS-6220

Code changes:
- Adds a `bulkWrite` function to the `dataService` so that import is not relying on the internals of `DataService`.
- Moves the `insertOne` handler to use `insertOne` from DataService instead of using _collection. 
I'll leave a comment on another change or two which typescript raised.

This PR helps prep for moving the import modal to LeafyGreen components. (hopefully the upcoming import/export epic as well). 

**Notes for reviewers**:
Be sure to open the [`src/modules/import.ts`](packages/compass-import-export/src/modules/import.ts) file, in github it's hidden by default to me.

### CODE REVIEW

File: packages/compass-import-export/src/modules/import.ts [hunk 1]
```diff
@@ -827,9 +875,9 @@ const reducer = (state = INITIAL_STATE, action) => {
 
   if (action.type === FINISHED) {
     const isComplete = state.status !== PROCESS_STATUS.CANCELED;
-    const hasErrors = state.errors.length > 0;
+    const hasErrors = (state.errors || []).length > 0;
 
-    let status = state.stats;
```
Comment (Anemy) [inline packages/compass-import-export/src/modules/import.ts:832]: Updated possible typo, doesn't seem likely to impact anything as a bug as this variable is pretty much always set to something else.

File: packages/compass-import-export/src/utils/collection-stream.ts [hunk 2]
```diff
@@ -67,27 +136,29 @@ class WritableCollectionStream extends Writable {
...
   }
 
-  async _executeBatch(callback) {
+  async _executeBatch(callback: (err?: Error) => void) {
     const documents = this.batch;
 
     this.batch = [];
 
-    let result;
-    let error;
```
Comment (Anemy) [inline packages/compass-import-export/src/utils/collection-stream.ts:79]: Looked unused, it was never set to anything.

File: packages/compass-import-export/src/utils/collection-stream.ts [hunk 3]
```diff
@@ -67,27 +136,29 @@ class WritableCollectionStream extends Writable {
...
+    let result: BulkOpResult;
 
     try {
-      result = await this._collection().bulkWrite(
-        documents.map((doc) => ({ insertOne: doc })),
+      result = await this.dataService.bulkWrite(
```
Comment (mcasimir) [inline packages/compass-import-export/src/utils/collection-stream.ts:150]: definitely not for this PR, but how does this gets canceled? Would it be worth creating a follow-up ticket in the import/export epic?

**Review by mcasimir — APPROVED** [review]

File: packages/data-service/src/data-service.ts [hunk 4]
```diff
@@ -645,6 +647,12 @@ export interface DataService {
     callback: Callback<Document | UpdateResult>
   ): void;
 
+  bulkWrite(
```
Comment (mcasimir) [inline packages/data-service/src/data-service.ts:650]: 👏 👏 👏

File: packages/compass-import-export/src/constants/file-types.ts [hunk 5]
```diff
@@ -1,4 +1,8 @@
-const FILE_TYPES = {
+export type AcceptedFileType = 'csv' | 'json';
+
+const FILE_TYPES: {
+  [AcceptedFileType: string]: string;
+} = {
```
Comment (addaleax) [inline packages/compass-import-export/src/constants/file-types.ts:5]: Just curious, this is making the type less specific than it previously was with just `as const` – any particular reason for that (now it’s `{ [key: string]: string }`, it used to be `{ CSV: 'csv', JSON: 'json' }`)?

**Review by addaleax — APPROVED** [review]

File: packages/compass-import-export/src/constants/file-types.ts (hunk 5 above)
Comment (Anemy) [reply packages/compass-import-export/src/constants/file-types.ts:5]: Nope, thanks for catching! Updated to be more strict:
```ts
const FILE_TYPES: {
  [fileType: string]: AcceptedFileType;
} = {
```
Eventually we'll remove this variable once everything is using the `AcceptedFileType`.

**Pull request merged**